packages/jsii-pacmak/lib/targets/dotnet.ts

```diff
@@ -338,38 +355,20 @@ class DotNetGenerator extends Generator {
             this.code.line('set;');
         }
         this.code.closeBlock();
-        this.code.line();
     }
 
     protected onBeginClass(cls: spec.ClassType, abstract: boolean) {
         let implementsExpr = '';
-        const baseNamespaces: string[] = [];
-        const baseTypeNames: string[] = [];
+        let baseNamespaces: string[] = [];
+        let baseTypeNames: string[] = [];
 
         const classBase = this.getClassBase(cls);
         const extendsExpression = classBase ? ` : ${classBase}` : '';
 
         if (cls.interfaces && cls.interfaces.length > 0) {
-
-            // For all base members
-            for (const base of cls.interfaces) {
-                // Retrieve the interface name from the fqn
-                const baseFqn = base.substr(0, base.lastIndexOf('.'));
-                const baseName = base.substr(base.lastIndexOf('.') + 1);
-                // Adding the base type
-                baseTypeNames.push(nameutils.convertInterfaceName(baseName));
-                if (baseFqn === this.assembly.name) {
-                    // If the base interface is in the current assembly
-                    // Nothing to do, we just added it to the list of implemented ifc
-                } else {
-                    // We need to add a reference to the interface assembly in the using statement and the csproj.
-                    const namespaceName = this.dependencies.get(baseFqn)!.namespace;
-                    // Adding the namespaceName to the base namespaces for the using statement
-                    if (!baseNamespaces.includes(namespaceName)) {
-                        baseNamespaces.push(namespaceName);
-                    }
-                }
-            }
+            const implemented: [string[], string[]] = this.resolveImplementedInterfaces(cls);
```

Comment:
const [baseTypeNamespaces, baseTypeNames] = this.resolveImplementedInterfaces(cls);

Reply:
Done